src/libs/state/BufferState.js

```diff
@@ -324,11 +324,15 @@ export default class BufferState {
         this.chathistory_request_count += 1;
         ircClient.chathistory[chathistoryFuncName](this.name, time)
             .then((event) => {
-                if (!event || event.commands.length === 0) {
+                if (!event) {
                     this.flag('chathistory_available', false);
-                } else {
-                    this.flag('chathistory_available', true);
+                    return;
                 }
+
+                let hasNewMessages = event.commands.some(msg =>
```

Comment:
This could do with a quick comment explaining what it's doing since it's just slightly out of context on its own here